File: 523.py
from typing import List


class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        mod_dict = {0: -1}
        sum_mod = [0] * (len(nums) + 1)
        for i in range(len(nums)):
            sum_mod[i + 1] = (sum_mod[i] + nums[i]
                              ) % k if k != 0 else sum_mod[i] + nums[i]
            if sum_mod[i + 1] in mod_dict:
                if i - mod_dict[sum_mod[i+1]] > 1:
                    return True
            else:
                mod_dict[sum_mod[i + 1]] = i
        return False


if __name__ == "__main__":
    s = Solution()
    print(s.checkSubarraySum([0, 0], 0))

File: 523.go
package leetcode

func checkSubarraySum(nums []int, k int) bool {
	pre := map[int]int{0: -1}
	sum := 0
	for i, val := range nums {
		if nums[i] == 0 && i > 0 && nums[i-1] == 0 {
			return true
		}
		sum = (sum + val) % k
		if ix, ok := pre[sum]; ok {
			if i-ix > 1 {
				return true
			}
		} else {
			pre[sum] = i
		}
	}
	return false
}
